src/material/badge/badge.ts

```diff
@@ -225,56 +202,52 @@ export class MatBadge extends _MatBadgeBase implements OnDestroy, OnChanges, Can
     return badgeElement;
   }
 
-  /** Sets the aria-label property on the element */
-  private _updateHostAriaDescription(newDescription: string, oldDescription: string): void {
-    // ensure content available before setting label
-    const content = this._updateTextContent();
+  /** Update the text content of the badge.  */
+  private _updateContent(newContent: string | number | undefined | null): void {
+    const newContentNormalized: string = `${newContent ?? ''}`.trim();
+    if (newContentNormalized && !this._badgeElement) {
+      this._badgeElement = this._createBadgeElement();
+    }
 
-    if (oldDescription) {
-      this._ariaDescriber.removeDescription(content, oldDescription);
+    if (this._badgeElement) {
+      this._badgeElement.textContent = newContentNormalized;
     }
 
+    this._content = newContentNormalized;
+    this._hasContent = !!newContentNormalized;
```

Comment:
Isn't this a little redundant considering that we can check `this._content`?

Reply:
Good point, now that it always normalizes the content this is unnecessary. Removed.